File: CoverPositionsCompile.php
<?php

$file = file_get_contents("_defaultConfigs/_newDefault.js");
$file = addslashes($file);
$file = str_replace("\r",'',$file);
$file = str_replace("\n",'\r\n',$file);

$file = '#define COVER_CONFIG_DEF_CONTENT "'.$file.'"';

file_put_contents("COVER_CONFIG_DEF_CONTENT.h",$file);


$configs = 'static const char* buildInCoverConfigs[] = {'."\\\r\n";

$count = 0;
$d = dir("_defaultConfigs");
while ($f = $d->read()){
   if ($f[0] == '.' || $f[0] == "_")
      continue;
   $file = file_get_contents("_defaultConfigs/".$f);
   $i = 0;
   while (ord($file[$i]) > 128) {
      $i++;
   }
   $content = addslashes(substr($file, $i));
   $content = str_replace("\r",'',$content);
   $content = str_replace("\n",'\r\n',$content);
   $content = str_replace("   ","\t",$content);
   $title = str_replace(".js","",$f).' (build-in)';
   $configs .= '"'.$title.'", "'.$content.'",  '."\\\r\n";
   $count++;
}

$configs .= '""};'."\r\n";

$configs .= 'static const int buildInCoverConfigCount = '.$count.";\r\n";

file_put_contents("BUILD_IN_COVERCONFIGS.h",$configs);


?>

File: CoverPositionsCompile.py
#!/bin/env python3

import json
import glob
import os.path

made_changes = False

def update_file_contents(fn, contents):
    """Set file contents, only writing if needed to reduce recompilation"""
    global made_changes
    with open(fn, "r+", newline="\n") as f:
        old_contents = f.read()
        if old_contents != contents:
            made_changes = True
            f.seek(0)
            f.truncate()
            f.write(contents)

with open("_defaultConfigs/_newDefault.js") as f:
    default_config = f.read()

update_file_contents(
    "COVER_CONFIG_DEF_CONTENT.h",
    '#define COVER_CONFIG_DEF_CONTENT {}\n'.format(
        json.dumps(default_config.replace("\n", "\r\n"))))

configs = {}
for fn in glob.glob("_defaultConfigs/*.js"):
    basename = os.path.basename(fn)
    if basename[0] == '_':
        continue
    with open(fn) as f:
        configs[basename[:-3]] = f.read()

out = """static const char* buildInCoverConfigs[] = {{
    {}
    ""
}};
static const int buildInCoverConfigCount = {};
""".format(
    "\n".join("{}, {},".format(json.dumps(k + " (build-in)"),
                               json.dumps(v.replace("\n", "\r\n").replace("   ", "\t")))
              for k, v in sorted(configs.items())),
    len(configs))

update_file_contents("BUILD_IN_COVERCONFIGS.h", out);

if made_changes:
    print("cover configs updated")
else:
    print("cover configs unchanged")
